src/lib/litegraph/src/LGraphNode.ts

```diff
@@ -845,15 +845,14 @@ export class LGraphNode
       }
 
       if (info.widgets_values) {
-        const widgetsWithValue = this.widgets.filter(
-          (w) => w.serialize !== false
-        )
-        for (let i = 0; i < info.widgets_values.length; ++i) {
-          const widget = widgetsWithValue[i]
-          if (widget) {
-            widget.value = info.widgets_values[i]
+        const widgetsWithValue = this.widgets
+          .values()
+          .filter((w) => w.serialize !== false)
+        widgetsWithValue.forEach((widget, i) => {
+          if (widget && i < info.widgets_values!.length) {
+            widget.value = info.widgets_values![i]
           }
-        }
+        })
```

Comment:
nit/optional:
When I see a conditional inside of a for loop, that feels like a missing filter.
Also, if there were nullish values in `widgets` wouldn't the original filter's `.serialize` throw?
Should there be a truthiness check first?

```suggestion
        const widgetsWithValue = this.widgets
          .values()
          .filter((w) => w.serialize !== false)
          .filter((widget, idx) => idx < info.widgets_values!.length)
        widgetsWithValue.forEach((widget, i) => {
          widget.value = info.widgets_values![i]
        })
```